### PR TITLE
Frame checking and validations

### PR DESCRIPTION
**Pull request by tiagovignatti**
Besides a small refactoring, this adds some pending frame checkings, frame bounds validation and size checking to guaranteeing the correct functioning as defined by
WebXrPresentationState. PTAL. Thank you very much!

### CODE REVIEW

File: wolvic/browser/vr/wvr_graphics_delegate.cc [hunk 1]
```diff
@@ -96,16 +96,21 @@ bool WvrGraphicsDelegate::CreateOrResizeWebXrSurface(
...
-  }
 
-  if (!j_surface_texture_) {
+    DCHECK(!j_surface_texture_);
     JNIEnv* env = base::android::AttachCurrentThread();
     j_surface_texture_ = Java_WVRSurfaceTexture_create(
         env,
         texture_handle_id_,
         webxr_surface_texture_.get()->j_surface_texture());
   }
```
Comment (svillar) [inline wolvic/browser/vr/wvr_graphics_delegate.cc:106]: leftover braces, I guess this does not compile?

File: wolvic/browser/vr/wvr_manager.cc [hunk 2]
```diff
@@ -499,6 +490,27 @@ WvrManager::GetInputSourceState() {
   return input_sources;
 }
 
+void WvrManager::DrawFrameSubmitNow() {
+  device::WebXrFrame* processing_frame = webxr_.GetProcessingFrame();
+  if (!SubmitFrameInternal(processing_frame->index))
+    return;
```
Comment (svillar) [inline wolvic/browser/vr/wvr_manager.cc:495]: We can pass the processing_frame as an argument and avoid calling GetProcessingFrame twice

Actually it's only used to call SubmitFrameInternal. To me this is an indication that these three lines should be on the caller and not here.

File: wolvic/browser/vr/wvr_manager.cc [hunk 3]
```diff
@@ -499,6 +490,27 @@ WvrManager::GetInputSourceState() {
...
+  // buffers.
+
+  // Renderer is waiting for the previous frame to render, unblock it now.
+  submit_client_->OnSubmitFrameRendered();
+
+  if (webxr_.HaveRenderingFrame())
+    webxr_.EndFrameRendering();
+  webxr_.TransitionFrameProcessingToRendering();
+
+  // See if we can animate a new WebVR frame.
```
Comment (svillar) [inline wolvic/browser/vr/wvr_manager.cc]: Typo: WebVR -> WebXR

**Review by svillar — CHANGES_REQUESTED** [review]
Looking good, just a few nits

File: wolvic/browser/vr/wvr_graphics_delegate.cc (hunk 1 above)
Comment (tiagovignatti) [reply wolvic/browser/vr/wvr_graphics_delegate.cc:106]: that was intentional. Please refer to the commit message: "It does also a small adjustment to create Java WVRSurfaceTexture only when the SurfaceTexture is also need to be created."

File: wolvic/browser/vr/wvr_manager.cc [hunk 4]
```diff
@@ -684,6 +735,21 @@ void WvrManager::UpdateLayerBounds(int16_t frame_index,
                                    const gfx::RectF& left_bounds,
                                    const gfx::RectF& right_bounds,
                                    const gfx::Size& source_size) {
+  if (!ValidateRect(left_bounds) || !ValidateRect(right_bounds)) {
```
Comment (jdapena) [inline wolvic/browser/vr/wvr_manager.cc:737]: How is it possible we get RectF with nan? Where are we getting this from?

Comment (tiagovignatti) [reply wolvic/browser/vr/wvr_manager.cc:737]: As the name suggests, `RectF` values are floating and may result in NaN depending the result of operations. Regardless, I have copied _ipsis litteris_ this snip from:
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/android/vr/gvr_scheduler_delegate.cc;l=71?q=ValidateRect

I'm seeing a similar usage of that for Wayland, for instance: 
https://source.chromium.org/chromium/chromium/src/+/main:ui/ozone/platform/wayland/host/wayland_frame_manager.cc?q=ValidateRect


Are you suggesting me to change something, sorry Dape?

File: wolvic/browser/vr/wvr_manager.cc (hunk 3 above)
Comment (tiagovignatti) [reply wolvic/browser/vr/wvr_manager.cc]: done.

File: wolvic/browser/vr/wvr_manager.cc (hunk 2 above)
Comment (tiagovignatti) [reply wolvic/browser/vr/wvr_manager.cc:495]: done.

**Comment by tiagovignatti** [comment]
Hi. I have just (force) pushed another iteration of this PR with the review points fixed. PTAL. Thank you.

File: wolvic/browser/vr/wvr_manager.cc (hunk 4 above)
Comment (jdapena) [reply wolvic/browser/vr/wvr_manager.cc:737]: No, no. Just making sure we do that check as early as we can. Is that the case?

Comment (tiagovignatti) [reply wolvic/browser/vr/wvr_manager.cc:737]: Yes.

**Review by zakharvoit — APPROVED** [review]

**Pull request merged**